tests/tools/test_file_tools.py

```diff
@@ -0,0 +1,125 @@
+import os
+import tempfile
+
+import pytest
+
+from promptulate.tools.file.toolkit import FileToolKit
+from promptulate.tools.file.tools import (
+    AppendFileTool,
+    CopyFileTool,
+    DeleteFileTool,
+    ListDirectoryTool,
+    MoveFileTool,
+    ReadFileTool,
+    WriteFileTool,
+)
+
+
+class TestFileToolKit:
+    # create a temporary directory
+    temp_dir = tempfile.TemporaryDirectory().name
+
+    def test_validate_root_dir(self):
+        # Test when root_dir is None,
+        # root_dir will be assigned to the current working directory
+        toolkit = FileToolKit(root_dir=None)
+        assert toolkit.root_dir == os.getcwd()
+
+        # Test when root_dir is an empty string,
+        # root_dir will be assigned to the current working directory
+        toolkit1 = FileToolKit(root_dir="")
+        assert toolkit1.root_dir == os.getcwd()
+
+        # Test when root_dir is a valid directory,
+        # root_dir will be assigned to the given directory
+        toolkit2 = FileToolKit(root_dir=self.temp_dir)
+        assert toolkit2.root_dir == self.temp_dir
+
+    def test_validate_tools(self):
+        # Test when mode_List is None,
+        # tools will be assigned to all tools
+        toolkit = FileToolKit(root_dir=self.temp_dir, modes=None)
+        tools = toolkit.get_tools()
+        assert len(tools) == 7
+
+        # Test when mode_List is an empty list,
+        # tools will be assigned to all tools
+        toolkit1 = FileToolKit(root_dir=self.temp_dir, modes=[])
+        tools = toolkit1.get_tools()
+        assert len(tools) == 7
+
+        # Tests that mode_List does not throw an exception
+        # when it contains an existing tool
+        toolkit2 = FileToolKit(root_dir=self.temp_dir, modes=["write", "read"])
+        assert toolkit2.modes == ["write", "read"]
+
+        # Tests that mode_List returns the corresponding tool
+        # when it contains an existing tool
+        toolkit3 = FileToolKit(root_dir=self.temp_dir, modes=["write", "read"])
+        tools = toolkit3.get_tools()
+        assert len(tools) == 2
+
+        # Test raises ValueError
+        # when mode_List contains a tool that does not exist
+        with pytest.raises(ValueError):
+            FileToolKit(root_dir=self.temp_dir, modes=["write", "nonexistent"])
+
+
+class TestFileTools:
+    # Create a temporary directory
+    temp_dir = tempfile.TemporaryDirectory()
+    dir_name = temp_dir.name
+    # Create a second temporary directory
+    temp_dir2 = tempfile.TemporaryDirectory()
+    dir_name2 = temp_dir2.name
+
+    # Create a file
+    temp_file = "test.txt"
+
+    def test_list_directory_tool(self):
+        # Test ListDirectoryTool's run method
+        Ltools = ListDirectoryTool(self.dir_name)
```

Comment:
You'd better assert Ltools.to_schema() == {xxx}. The following tools are the same.